src/main/java/com/uber/profiling/AgentImpl.java

```diff
@@ -76,7 +80,28 @@ public void run(Arguments arguments, Instrumentation instrumentation, Collection
 
         if (!arguments.getDurationProfiling().isEmpty()
                 || !arguments.getArgumentProfiling().isEmpty()) {
-            instrumentation.addTransformer(new JavaAgentFileTransformer(arguments.getDurationProfiling(), arguments.getArgumentProfiling()));
+            instrumentation.addTransformer(new JavaAgentFileTransformer(arguments.getDurationProfiling(),
+                    arguments.getArgumentProfiling()), true);
+
+            Set<String> loadedClasses = Arrays.stream(instrumentation.getAllLoadedClasses())
+                    .map(Class::getName).collect(Collectors.toSet());
+
+            Set<String> tobeReloadClasses = arguments.getDurationProfiling().stream()
+                    .map(ClassAndMethod::getClassName).collect(Collectors.toSet());
+
+            tobeReloadClasses.addAll(arguments.getArgumentProfiling().stream()
+                    .map(ClassMethodArgument::getClassName).collect(Collectors.toSet()));
+
+            tobeReloadClasses.retainAll(loadedClasses);
+
+            tobeReloadClasses.forEach(clazz -> {
+                try {
+                    instrumentation.retransformClasses(Class.forName(clazz));
+                    logger.info("Reload class [" + clazz + "] success.");
+                } catch (Exception e) {
+                    logger.warn("Reload class [" + clazz + "] failed.");
```

Comment:
shall we pass and print out the exception in waring logging also?
like `logger.warn("Reload class [" + clazz + "] failed.", e);`

Reply:
fixed it, thanks.